### PR TITLE
ui update, add debug panel

### PR DESCRIPTION
**Pull request by mdroidian**

### CODE REVIEW

File: src/components/DailyConfigComponent.tsx [hunk 1]
```diff
@@ -18,41 +19,51 @@ import localStorageSet from "roamjs-components/util/localStorageSet";
 import nanoid from "nanoid";
 
 const DailyConfig = () => {
-  // TODO refresh config so debug panel shows up to date info
   const config = useMemo(getDailyConfig, []);
```
Comment (dvargas92495) [inline src/components/DailyConfigComponent.tsx:22]: `initialConfig`

File: src/components/DailyConfigComponent.tsx [hunk 2]
```diff
@@ -18,41 +19,51 @@ import localStorageSet from "roamjs-components/util/localStorageSet";
...
+    enabled,
+    device,
+    time: setRunTime,
+    "workflow name": workflowName,
+    "last-run": lastRun,
+    "next-run": configNextRun,
+  } = localConfig;
   const [onlyOnThisDevice, setOnlyOnThisDevice] = useState(
-    !!config.device && config.device === localStorageGet("device")
+    !!device && device === localStorageGet("device")
```
Comment (dvargas92495) [inline src/components/DailyConfigComponent.tsx:33]: Id move `localStorageGet("device")` to `getDailyConfig` defaults

File: src/components/DailyConfigComponent.tsx [hunk 3]
```diff
@@ -18,41 +19,51 @@ import localStorageSet from "roamjs-components/util/localStorageSet";
...
   );
-  const [workflowName, setWorkflowName] = useState(config["workflow name"]);
+
   const defaultTime = useMemo(() => {
     const date = new Date();
-    if (config && config["time"]) {
-      const [h, m] = config["time"].split(":").map(Number);
+    if (localConfig && setRunTime) {
```
Comment (dvargas92495) [inline src/components/DailyConfigComponent.tsx:38]: please remove `set` lol it's got special meaning in React

File: src/components/DailyConfigComponent.tsx [hunk 4]
```diff
@@ -18,41 +19,51 @@ import localStorageSet from "roamjs-components/util/localStorageSet";
...
       date.setMinutes(m);
     } else {
       date.setHours(0);
       date.setMinutes(0);
     }
     return date;
-  }, [config]);
-  const lastRun = config?.["last-run"];
+  }, [localConfig]);
+  const [localTime, setLocalTime] = useState(defaultTime);
```
Comment (dvargas92495) [inline src/components/DailyConfigComponent.tsx:48]: Id centralize with `localConfig.time`

File: src/components/DailyConfigComponent.tsx [hunk 5]
```diff
@@ -18,41 +19,51 @@ import localStorageSet from "roamjs-components/util/localStorageSet";
 import nanoid from "nanoid";
 
 const DailyConfig = () => {
-  // TODO refresh config so debug panel shows up to date info
   const config = useMemo(getDailyConfig, []);
-  const [disabled, setDisabled] = useState(!config.enabled);
+  const [localConfig, setLocalConfig] = useState(config);
```
Comment (dvargas92495) [inline src/components/DailyConfigComponent.tsx:23]: you may want a:
```typescript
const [localConfig, _setLocalConfig] = useState(config);
const setLocalConfig = useCallback((newStuff) => {
    _setLocalConfig((prev) => {
        return ({...prev, ...newStuff})
    });
    saveDailyConfig(newStuff);
}, [_setLocalConfig]);
```

**Review by dvargas92495 — COMMENTED** [review]
✅ after state consolidation

File: src/components/DailyConfigComponent.tsx (hunk 2 above)
Comment (mdroidian) [reply src/components/DailyConfigComponent.tsx]: I couldn't get localStorageGet to work in zDailyConfig.

I tried
`currentDevice: localStorageGet("device"),`
`currentDevice: z.string().optional().default(""),`
and
`const currentDevice = localStorageGet("device");`
``currentDevice: currentDevice,``

File: src/components/DailyConfigComponent.tsx (hunk 4 above)
Comment (mdroidian) [reply src/components/DailyConfigComponent.tsx]: changed to `timePicker` to reflect that it's used in that component.  I think this is required unless we call setDailyConfig on each HH/MM change.

File: src/components/DailyConfigComponent.tsx [hunk 6]
```diff
@@ -51,91 +72,171 @@ const DailyConfig = () => {
...
+                      className={"user-select-none flex-1 text-center"}
+                    />
+                    <Icon
+                      icon={isEditTime ? "tick" : "edit"}
+                      onClick={async () => {
+                        if (isEditTime) {
+                          const newTime = `${timePicker.getHours()}:${timePicker.getMinutes()}`;
+                          await setConfig({ time: newTime });
+                          await scheduleNextDailyRun({ tomorrow: true });
+                          _setLocalConfig(getDailyConfig()); // force update, scheduleNextDailyRun runs saveDailyConfig
```
Comment (mdroidian) [inline src/components/DailyConfigComponent.tsx:124]: This doesn't seem ideal.

File: src/components/DailyConfigComponent.tsx [hunk 7]
```diff
@@ -51,91 +72,171 @@ const DailyConfig = () => {
...
+        )}
+        <Tabs.Expander />
+        <Switch
+          large={true}
+          className="w-full text-right"
+          checked={enabled}
+          onChange={async (e) => {
+            const enabled = (e.target as HTMLInputElement).checked;
+            if (enabled) {
+              await setConfig({
```
Comment (mdroidian) [inline src/components/DailyConfigComponent.tsx:225]: this `await` might mitigate some users problems

https://www.loom.com/share/bca4d70c4f1546478659b81667a5e55c

File: src/components/DailyConfigComponent.tsx (hunk 6 above)
Comment (mdroidian) [reply src/components/DailyConfigComponent.tsx:124]: Running into some type errors up the chain

adding `Promise<DailyConfig>` to `scheduleNextDailyRun` and `getDailyConfig` still left `getExtensionApi().settings.set` returning `void`, so left it as is

**Pull request merged**